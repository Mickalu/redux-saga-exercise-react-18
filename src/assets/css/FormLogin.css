.group_button_form {
  display: flex;
  justify-content: flex-start;
}

.row_form__button {
  margin-left: 0;
  margin-right: 0;
}
