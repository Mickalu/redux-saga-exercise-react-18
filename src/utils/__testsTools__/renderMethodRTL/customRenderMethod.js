import React from "react";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";

import { store } from "../../../store";

const getByTextCustom = (text) => (
  screen.getByText(text)
);

const queryAllByRoleCustom = (role, options) => (
  screen.queryAllByRole(role, options)
);

const queryByTestIdCustom = (testId) => (
  screen.queryByTestId(testId)
);

const getByTestIdCustom = (testId) => (
  screen.getByTestId(testId)
);

const getByDisplayValueCustom = (displayValue) => (
  screen.getByDisplayValue(displayValue)
);

const renderWithProviders = ({ children }) => (
  <Provider store={store}>
    { children }
  </Provider>
);

export const renderCustom = (ui, options = {}) => render(ui, { wrapper: renderWithProviders, ...options });

// re-export everything
export * from '@testing-library/react';

export { renderCustom as render };
export { getByTextCustom as getByText };
export { queryAllByRoleCustom as queryAllByRole };
export { queryByTestIdCustom as queryBytestId };
export { getByTestIdCustom as getByTestId };
export { getByDisplayValueCustom as getByDisplayValue };
