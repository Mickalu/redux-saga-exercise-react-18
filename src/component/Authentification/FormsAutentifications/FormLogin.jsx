import React from 'react';
import { Link } from 'react-router-dom';

import Col from 'react-bootstrap/Col';
import Button from 'react-bootstrap/Button';
import Form from "react-bootstrap/Form";
import Row from 'react-bootstrap/Row';
import "../../../assets/css/FormLogin.css";

const FormLogin = ({ submitUserConnection, changeValueUsername, changeValuePassword, error_message }) => (
  <Form onSubmit={submitUserConnection}>
    <Row>
      <Form.Group className="mb-3">
        <Form.Label>Username : </Form.Label>
        <Form.Control onChange={(e) => changeValueUsername(e.target.value)} type="text" />
      </Form.Group>

      <Form.Group className="mb-3">
        <Form.Label>Password : </Form.Label>
        <Form.Control onChange={(e) => changeValuePassword(e.target.value)} type="password" placeholder="Password" />
      </Form.Group>
    </Row>

    <Row>

    </Row>
      {error_message()}
    <Row>
      <Col className='row_form__button'>
        <Button variant="primary" type="submit">
            Submit
        </Button>
      </Col>

      <Col className='row_form__button register_button'>
        <Link to="register">
          <Button variant='link'>
            Register
          </Button>
        </Link>
      </Col>
    </Row>
  </Form>
);

export default FormLogin
