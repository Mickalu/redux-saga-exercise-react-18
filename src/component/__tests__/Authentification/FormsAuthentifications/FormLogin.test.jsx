import React from "react";

import { render } from "../../../../utils/__testsTools__/renderMethodRTL/customRenderMethod";
import FormLogin from "../../../Authentification/FormsAutentifications/FormLogin";

it("FormLogin should match snapshot", () => {
  const { container } = render(<FormLogin error_message={jest.fn()} />);
  expect(container).toMatchSnapshot();
});
