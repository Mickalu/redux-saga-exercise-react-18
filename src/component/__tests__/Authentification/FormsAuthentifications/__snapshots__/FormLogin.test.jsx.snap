// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLogin should match snapshot 1`] = `
<div>
  <form
    class=""
    data-testid="form-login"
  >
    <div
      class="row"
    >
      <div
        class="mb-3"
      >
        <label
          class="form-label"
        >
          Username : 
        </label>
        <input
          class="form-control"
          type="text"
        />
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
        >
          Password : 
        </label>
        <input
          class="form-control"
          placeholder="Password"
          type="password"
        />
      </div>
    </div>
    <div
      class="row"
    />
    <div
      class="row"
    >
      <div
        class="row_form__button col"
      >
        <button
          class="btn btn-primary"
          type="submit"
        >
          Submit
        </button>
      </div>
      <div
        class="row_form__button register_button col"
      >
        <a
          href="/register"
        >
          <button
            class="btn btn-link"
            type="button"
          >
            Register
          </button>
        </a>
      </div>
    </div>
  </form>
</div>
`;
